src/snowflake/connector/auth/oauth_code.py

```diff
@@ -385,3 +396,27 @@ def _parse_authorization_redirected_request(
                 },
             )
         return parsed.get("code", [None])[0], parsed.get("state", [None])[0]
+
+    def _eligible_for_default_client_credentials(
+        self,
+        client_id: str,
+        client_secret: str,
+        authentication_url: str,
+        token_request_url: str,
+        host: str,
+    ) -> bool:
+        return (
+            client_id == ""
+            and client_secret == ""
+            and self.__class__._is_snowflake_as_idp(
```

Comment:
Should this be possible to override in subclasses?
If not then just a nit:
```suggestion
            and AuthByOauthCode._is_snowflake_as_idp(
```

Reply:
I think that if for some reason a subclass overrides this function it should be reflected in this behaviour, without a need to override also the other methods.